INTODUCTION

Placement status prediction using machine learning it aims to analyze student data to predict their likelihood of securing a job placement based on factors like academic performance, skills, and extracurricular activities.

IMPORT LIBRARIES AND LOADING DATA

In [49]:
import pandas as pd 
import numpy as np
import warnings
warnings.filterwarnings('ignore')
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import StandardScaler,LabelEncoder
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
# from sklearn.ensemble import AdaBoostClassifier
from sklearn import metrics
data=pd.read_csv(r'C:\Users\Lenovo\Documents\Data analytics\data_analytics\project\placementdata.csv')
data.head()

StudentID  CGPA  Internships  Projects  Workshops/Certifications  \
0          1   7.5            1         1                         1   
1          2   8.9            0         3                         2   
2          3   7.3            1         2                         2   
3          4   7.5            1         1                         2   
4          5   8.3            1         2                         2   

   AptitudeTestScore  SoftSkillsRating ExtracurricularActivities  \
0                 65               4.4                        No   
1                 90               4.0                       Yes   
2                 82               4.8                       Yes   
3                 85               4.4                       Yes   
4                 86               4.5                       Yes   

  PlacementTraining  SSC_Marks  HSC_Marks PlacementStatus  
0                No         61         79       NotPlaced  
1               Yes         78         82          Placed  
2                No         79         80       NotPlaced  
3               Yes         81         80          Placed  
4               Yes         74         88          Placed

DATA CLEANING

In [50]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   StudentID                  10000 non-null  int64  
 1   CGPA                       10000 non-null  float64
 2   Internships                10000 non-null  int64  
 3   Projects                   10000 non-null  int64  
 4   Workshops/Certifications   10000 non-null  int64  
 5   AptitudeTestScore          10000 non-null  int64  
 6   SoftSkillsRating           10000 non-null  float64
 7   ExtracurricularActivities  10000 non-null  object 
 8   PlacementTraining          10000 non-null  object 
 9   SSC_Marks                  10000 non-null  int64  
 10  HSC_Marks                  10000 non-null  int64  
 11  PlacementStatus            10000 non-null  object 
dtypes: float64(2), int64(7), object(3)
memory usage: 937.6+ KB


In [51]:
data.isna().sum()

StudentID                    0
CGPA                         0
Internships                  0
Projects                     0
Workshops/Certifications     0
AptitudeTestScore            0
SoftSkillsRating             0
ExtracurricularActivities    0
PlacementTraining            0
SSC_Marks                    0
HSC_Marks                    0
PlacementStatus              0
dtype: int64

In [52]:
data['PlacementStatus'].unique()

array(['NotPlaced', 'Placed'], dtype=object)

In [53]:
le=LabelEncoder()
data['PlacementStatus']=le.fit_transform(data['PlacementStatus'])
data['ExtracurricularActivities']=le.fit_transform(data['ExtracurricularActivities'])
data['PlacementTraining']=le.fit_transform(data['PlacementTraining'])

EXTRACTING INDEPENDENT AND DEPENDENT VARIBLE

In [54]:
x=data.iloc[:,1:11]
x=pd.DataFrame(x)
y=data['PlacementStatus']
y=pd.DataFrame(y)

SPLITING DATASET INTO TRAIN AND TEST DATA

In [55]:
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=.25,random_state=46)

FEATURE SCALING

In [56]:
st_x=StandardScaler()
x_test=st_x.fit_transform(x_test)
x_train=st_x.fit_transform(x_train)

MODELS BUILDING

CHECKING ACCURACY BY USING LOGISTIC REGRESSION

In [57]:
model=LogisticRegression()
model.fit(x_train,y_train)

LogisticRegression()

In [58]:
y_predict=model.predict(x_test)
print(y_test)
print(y_predict)

      PlacementStatus
1909                1
4189                1
5819                0
8449                1
6501                1
...               ...
6716                1
8383                1
9611                0
9142                0
2695                0

[2500 rows x 1 columns]
[0 1 0 ... 0 1 0]


In [59]:
print('mse value logistics:',metrics.mean_squared_error(y_test,y_predict))
print('accuracy logistics:',metrics.accuracy_score(y_test,y_predict))

mse value logistics: 0.1912
accuracy logistics: 0.8088


CHECKING ACCURACY BY USING KNN

In [60]:
from sklearn.neighbors import KNeighborsClassifier
classifier=KNeighborsClassifier(n_neighbors=5,metric='minkowski',p=2)
classifier.fit(x_train,y_train)

KNeighborsClassifier()

In [61]:
y_predict=classifier.predict(x_test)
print(y_test)
print(y_predict)

      PlacementStatus
1909                1
4189                1
5819                0
8449                1
6501                1
...               ...
6716                1
8383                1
9611                0
9142                0
2695                0

[2500 rows x 1 columns]
[0 0 0 ... 0 1 0]


In [62]:
print("mse value of KNN:",metrics.mean_absolute_error(y_predict,y_test))
print('accuracy in KNN:',metrics.accuracy_score(y_predict,y_test))

mse value of KNN: 0.2152
accuracy in KNN: 0.7848


CHECKING ACCURACY BY USING RANDOM FOREST

In [63]:
from sklearn.ensemble import RandomForestClassifier
model2=RandomForestClassifier(n_estimators=10,criterion='entropy')
model2.fit(x_train,y_train)


RandomForestClassifier(criterion='entropy', n_estimators=10)

In [64]:
y_predict=model2.predict(x_test)
print(y_test)
print(y_predict)

      PlacementStatus
1909                1
4189                1
5819                0
8449                1
6501                1
...               ...
6716                1
8383                1
9611                0
9142                0
2695                0

[2500 rows x 1 columns]
[0 0 0 ... 0 1 0]


In [65]:
print("mse value of random forest:",metrics.mean_absolute_error(y_predict,y_test))
print('accuracy in random forest:',metrics.accuracy_score(y_predict,y_test))


mse value of random forest: 0.2256
accuracy in random forest: 0.7744


CHECKING ACCURACY BY USING SVM

In [66]:
from sklearn.svm import SVC
model3=SVC(kernel='linear')
model3.fit(x_train,y_train)

SVC(kernel='linear')

In [67]:
y_predict=model3.predict(x_test)
print(y_test)
print(y_predict)

      PlacementStatus
1909                1
4189                1
5819                0
8449                1
6501                1
...               ...
6716                1
8383                1
9611                0
9142                0
2695                0

[2500 rows x 1 columns]
[0 1 0 ... 0 1 0]


In [68]:
print("mse value of svm:",metrics.mean_absolute_error(y_predict,y_test))
print('accuracy in svm:',metrics.accuracy_score(y_predict,y_test))


mse value of svm: 0.188
accuracy in svm: 0.812


CHECKING ACCURACY BY USING DECISION TREE

In [69]:
from sklearn.tree import DecisionTreeClassifier
model4=DecisionTreeClassifier(criterion='entropy',random_state=67)
model4.fit(x_train,y_train)

DecisionTreeClassifier(criterion='entropy', random_state=67)

In [70]:
y_predict=model4.predict(x_test)
print(y_test)
print(y_predict)

      PlacementStatus
1909                1
4189                1
5819                0
8449                1
6501                1
...               ...
6716                1
8383                1
9611                0
9142                0
2695                0

[2500 rows x 1 columns]
[0 0 0 ... 0 1 0]


In [71]:
print("mse value of decision tree:",metrics.mean_absolute_error(y_predict,y_test))
print('accuracy in decision tree:',metrics.accuracy_score(y_predict,y_test))


mse value of decision tree: 0.2824
accuracy in decision tree: 0.7176


SUMMARY

In this project, we use machine learning algorithms to predict a student’s chances of getting placed in a job after graduation. After testing various models,SVM provided the best results with accuracy of 81.2%, so making it ideal for predicting placement outcomes.